src/services/console/warn_once.test.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { warnOnce } from './warn_once';
+
+describe('warnOnce', () => {
+  const warn = jest.spyOn(console, 'warn').mockImplementation(() => {});
```

Comment:
Rely on jest's lifecycle hooks for more specific management, otherwise this test's execution is non-deterministic and dependent on the order other tests are run.

```suggestion
  let warn;
  beforeAll(() => {
    warn = jest.spyOn(console, 'warn').mockImplementation(() => {});
  })
```